src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -454,16 +455,25 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             mir::PlaceRef {
                 base: mir::PlaceBase::Static(box mir::Static {
                     ty,
-                    kind: mir::StaticKind::Promoted(promoted),
+                    kind: mir::StaticKind::Promoted(promoted, substs),
+                    def_id,
                 }),
                 projection: None,
             } => {
+                debug!("promoted={:?}, def_id={:?}, substs={:?}, self_substs={:?}", promoted, def_id, substs, self.instance.substs);
                 let param_env = ty::ParamEnv::reveal_all();
+                let instance = Instance::new(*def_id, substs.subst(bx.tcx(), self.instance.substs));
```

Comment:
I tried that but I was getting an ICE because we need to use the substituted `substs` and `monomorphize()` is hard coded to use `self.instance.substs`.

Reply:
How did you call it? I'd assume you'd do `self.monomorphize(substs)`

Reply:
Yeah, you're right. I was thinking of the call to `self.monomorphize(&ty)` that I changed. That call needs to use `instance.substs` which can be different than `self.instance.substs`.